### PR TITLE
Unused Stapler Facet is removed

### LINKED ISSUE #87: Is Stapler Facet even needed?
### What feature do you want to see added?

Stapler Facet is empty. It does not contain any framework-specific settings that need to be configured on the module.
It is essentially just a toggle for a few features. 
Is it even necessary?

### Upstream changes

_No response_

### PR DESCRIPTION
**Pull request by jgreffe**
Fixes #87 

Remove all classes related to Facet
Remove all plugin.xml declarations related to Facet
Fixes infinite loop in `JexlInspection.tokenize("'")` : case `${%Test('`
Fixes out of range issue with patterns like `${%Test(something)}` (`Argument rangeInElement (20,21) endOffset must not exceed descriptor text range (121, 141) length (20)`)

Upgrading to this new plugin version on existing projects having facet will popup a warning message:
Monosnap Error 2022-11-02 12-50-04

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by duemir** [comment]
@timja I am on the fence about this. What do you think?

**Comment by timja** [comment]
Sounds fine to remove if it’s not being used, but not too familiar with this

**Comment by duemir** [comment]
OK. We should probably do it. It is unlikely that this Facet was ever used in the field by anybody but me. 
But I think we should add a few more tests for JexlInspection, which is being published to a wider audience by this change. e.g., test that it handles property lookups (l10n) like `${%Name}` and `${%Build Queue(items.size())}`

**Comment by jgreffe** [comment]
Added some unit tests, and fixed two issues:
- infinite loop in JexlInspection.tokenize("'")
- out of range issue with patterns like `${%Test(something)}` (Argument rangeInElement (20,21) endOffset must not exceed descriptor text range (121, 141) length (20))

File: src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java [hunk 1]
```diff
@@ -43,4 +35,48 @@ public void testParserFailureJexlInspection() {
...
+        };
+
+        JexlInspection jexlInspection = new JexlInspection();
+
+        for (Map.Entry<String, String> expectation : expectations.entrySet()) {
+            assertEquals("Issue with text: " + expectation.getKey(), expectation.getValue(),
+                         jexlInspection.tokenize(expectation.getKey()));
+        }
+    }
+};
```
Comment (duemir) [inline src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:82]:
```suggestion
}
```

File: src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java [hunk 2]
```diff
@@ -43,4 +35,48 @@ public void testParserFailureJexlInspection() {
...
+        myFixture.enableInspections(new JexlInspection());
+        List<HighlightInfo> highlightInfos = myFixture.doHighlighting(HighlightSeverity.WEAK_WARNING);
+        assertNotEmpty(highlightInfos);
+        assertEquals("Expected error", 2, highlightInfos.size());
+        assertEquals("Missing '}' character at the end of expression", highlightInfos.get(0).getDescription());
+        assertTrue("Should match error", highlightInfos.get(1).getDescription().contains("Expecting \"(\" ..."));
+    }
+
+    public void testTokenize() {
+        Map<String, String> expectations = new HashMap<String, String>() {
```
Comment (duemir) [inline src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:50]:
```suggestion
        Map<String, String> expectations = new HashMap<>() {
```

File: src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java [hunk 3]
```diff
@@ -11,24 +11,16 @@
 import com.intellij.testFramework.fixtures.DefaultLightProjectDescriptor;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
```
Comment (duemir) [inline src/test/java/org/kohsuke/stapler/idea/JexlInspectionTest.java:16]: imports need to be cleaned up.

**Review by duemir — CHANGES_REQUESTED** [review]
A few minor suggestions. 

I've confirmed the test fail and that changes in Jexl Inspection fix them. I have a hard time reasoning about this thing, so this is as best I can do in a limited time.

**Review by duemir — APPROVED** [review]

**Comment by duemir** [comment]
Oh boy! I built a local version of the next release. Ran the JEXL inspection on @jenkinsci/jenkins . It crashes in two files and shows 11 errors in 9 files. 

4 has to do with localization. The rest is about HTML-encoded angle bracket 🤔

**Comment by jgreffe** [comment]
@duemir : this means as jexl inspection is now activated by default, it displays proper errors, but also point out on faulty plugin code (crashes) ?

**Comment by duemir** [comment]
@jgreffe Yes, as far as I can see. I will try to iron at least some out.

**Comment by jgreffe** [comment]
Crashes occur with i18n like:
```
<j:jelly xmlns:j="jelly:core">
    <p>
        ${%Te'st}</p>
</j:jelly>
```

**Pull request merged**